src/modules/users/test/users.service.spec.ts

```diff
@@ -133,4 +134,51 @@ describe('UsersService', () => {
       );
     });
   });
+
+  describe('wishlistAccommodation', () => {
+    it('should add accommodation to wishlist if not already in', async () => {
+      const userId = 'existing-user-id';
+      const accommodationId = 'accommodation-2';
+
+      const updatedUser = await service.wishlistAccommodation(
+        userId,
+        accommodationId,
+      );
+
+      expect(updatedUser.wishlist).toContain(accommodationId);
+      expect(usersRepository.save).toHaveBeenCalledWith(
+        expect.objectContaining({
+          wishlist: expect.arrayContaining([accommodationId]),
+        }),
+      );
+    });
+
+    it('should remove accommodation from wishlist if already in', async () => {
+      const userId = 'existing-user-id';
+      const accommodationId = 'accommodation-1';
+
+      const updatedUser = await service.wishlistAccommodation(
+        userId,
+        accommodationId,
+      );
+
+      expect(updatedUser.wishlist).not.toContain(accommodationId);
+      expect(usersRepository.save).toHaveBeenCalledWith(
+        expect.objectContaining({
+          wishlist: expect.arrayContaining([]),
+        }),
+      );
+    });
+
+    it('should throw NotFoundException if user does not exist', async () => {
+      const userId = 'non-existing-user-id';
+      const accommodationId = 'accommodation-2';
+
+      await expect(
```

Comment:
I think you can also omit await here and it will still work.

Reply:
it can work, but it is connection with database part, so I added async await